tests/integration/integration_test.go

```diff
@@ -525,16 +526,53 @@ func TestDestroyStackRef(t *testing.T) {
 	}()
 
 	e.ImportDirectory("large_resource/nodejs")
-	e.RunCommand("pulumi", "login", "--cloud-url", e.LocalURL())
+	organization := "organization"
+	if useLocal {
+		if !fullyQualifiedRefs {
+			t.Setenv("PULUMI_SELF_MANAGED_STATE_LEGACY_LAYOUT", "true")
+		}
+		e.RunCommand("pulumi", "login", "--cloud-url", e.LocalURL())
+	} else {
+		require.True(t, fullyQualifiedRefs, "fullyQualifiedRefs must be true when using the Pulumi Service")
+		if os.Getenv("PULUMI_ACCESS_TOKEN") == "" {
+			t.Skipf("Skipping: PULUMI_ACCESS_TOKEN is not set")
+		}
+		output, _ := e.RunCommand("pulumi", "whoami")
+		organization = strings.TrimSpace(output)
+	}
 
-	e.RunCommand("pulumi", "stack", "init", "dev")
+	stackName, err := resource.NewUniqueHex("rm-test-", 8, -1)
+	contract.AssertNoErrorf(err, "resource.NewUniqueHex should not fail with no maximum length is set")
+
+	e.RunCommand("pulumi", "stack", "init", stackName)
 
 	e.RunCommand("yarn", "link", "@pulumi/pulumi")
 	e.RunCommand("yarn", "install")
 
 	e.RunCommand("pulumi", "up", "--skip-preview", "--yes")
 	e.CWD = os.TempDir()
-	e.RunCommand("pulumi", "destroy", "--skip-preview", "--yes", "-s", "organization/large_resource_js/dev")
+	stackRef := stackName
+	if fullyQualifiedRefs {
+		stackRef = organization + "/large_resource_js/" + stackName
+	}
+
+	e.RunCommand("pulumi", "destroy", "--skip-preview", "--yes", "-s", stackRef)
+	e.RunCommand("pulumi", "stack", "rm", "--yes", "-s", stackRef)
+}
+
+//nolint:paralleltest // uses parallel programtest
+func TestDestroyStackRef_LocalProject(t *testing.T) {
+	testDestroyStackRef(t, true, true)
+}
+
+//nolint:paralleltest // uses parallel programtest
+func TestDestroyStackRef_LocalNonProject(t *testing.T) {
+	testDestroyStackRef(t, true, false)
+}
+
+//nolint:paralleltest // uses parallel programtest
+func TestDestroyStackRef_Cloud(t *testing.T) {
+	testDestroyStackRef(t, false, true)
```

Comment:
If we're doing separate positional bools here, annotating them would be valuable:

```suggestion
	testDestroyStackRef(t, true /* local */, true /* qualified */)
}

//nolint:paralleltest // uses parallel programtest
func TestDestroyStackRef_LocalNonProject(t *testing.T) {
	testDestroyStackRef(t, true /* local */, false /* qualified */)
}

//nolint:paralleltest // uses parallel programtest
func TestDestroyStackRef_Cloud(t *testing.T) {
	testDestroyStackRef(t, false /* local */, true /* qualified */)
```